### PR TITLE
Updated Test For Connectors

### PR DESCRIPTION
**Pull request by adiyaar24**
## Describe your changes
1. Data Sources Test For Connectors 
2. SSH Tests Added for Resources
3. Role Assignment Doc Updation

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

**Comment by rathodmeetsatish** [comment]
trigger build

**Comment by rathodmeetsatish** [comment]
trigger subcategoryfieldcheck

File: .changelog/509.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+```release-note:enhancement
+resources/platform_role_assignments: Make resource_group_identifier , role_identifier and type under principle schema required.
+resources/platform_secret_sshkey: Updated the Behaviour of referening the secrets at account,project and org Level . Made Key Field Required in SSH credential of type keyReference
```
Comment (rijajoo) [inline .changelog/509.txt:3]: Lots of typos. Please fix this.

File: internal/service/platform/connector/vault_test.go [hunk 2]
```diff
@@ -1042,7 +1043,7 @@ func testOrgResourceConnectorVault_aws_auth(id string, name string, vault_secret
 
 	resource "time_sleep" "wait_4_seconds" {
 		depends_on = [harness_platform_secret_text.test]
-		create_duration = "4s"
+		create_duration = "10s"
```
Comment (rijajoo) [inline internal/service/platform/connector/vault_test.go:1046]: Why are we updating the time here? Also why is it still called wait_4_seconds? This is misleading.

**Review by rijajoo — CHANGES_REQUESTED** [review]

File: .changelog/509.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+```release-note:enhancement
+resources/platform_role_assignments: Made resource_group_identifier, role_identifier and type under principle schema required.
```
Comment (rijajoo) [inline .changelog/509.txt:2]: principal*

**Review by rijajoo — APPROVED** [review]

**Review by rijajoo — APPROVED** [review]

**Review by rijajoo — APPROVED** [review]

**Pull request merged**